### DISCUSSION TITLE
UNet diffrent stage output

### DISCUSSION THREAD

**Discussion by YingdiZhang** [Project-MONAI/MONAI#6386] [Q&A]
Thanks for the great job. As I know, there are 4 downsample stage in UNet. How to output the four stage feature separately.

**Comment by KumoLiu** [comment]
Hi @YingdiZhang, maybe you could refer to this [discussion](https://github.com/Project-MONAI/MONAI/discussions/5796).

Hope it can help you, thanks!

**Reply by YingdiZhang** [reply]
Thanks for your reply, it seems I should input an argument which is the name of intermediate layer for example "last_layer". While after reading the source code of UNet, I dont find the four stage intermediate layer's name.

**Reply by KumoLiu** [reply]
Hi @YingdiZhang, "last_layer" is just a key used in "intermediate_outputs". You could see the whole model architecture by using `print(net)`. You could also get the intermediate output like `print(net.model[1].submodule[2])`.

Hope it can help you, thanks!
